seaborn/matrix.py

```diff
@@ -964,18 +982,49 @@ def plot_colors(self, xind, yind, **kws):
         if self.row_colors is not None:
             matrix, cmap = self.color_list_to_matrix_and_cmap(
                 self.row_colors, yind, axis=0)
+
+            # Get labels from colors if given as dataframe
+            if isinstance(self.row_colors, pd.DataFrame):
+                xticklabels = self.row_colors.columns
+            elif isinstance(self.row_colors, pd.Series) and \
+                    self.row_colors.name:
+                xticklabels = [self.row_colors.name]
+            else:
+                xticklabels = False
+
             heatmap(matrix, cmap=cmap, cbar=False, ax=self.ax_row_colors,
-                    xticklabels=False, yticklabels=False,
+                    xticklabels=xticklabels, yticklabels=False,
                     **kws)
+
+            # Adjust rotation of labels
+            if xticklabels is not False:
+                self.ax_row_colors.set_xticklabels(
+                    self.ax_row_colors.get_xticklabels(), rotation=90)
```

Comment:
I think that's reasonable...I don't think it should be configurable, but it would be nice if it were easier to predict the relative size of matplotlib text and other elements to automatically choose a good orientation. But I don't think it needs to be handled here.